core/src/main/scala/kafka/tools/ConsumerPerformance.scala

```diff
@@ -184,12 +212,23 @@ object ConsumerPerformance {
                            lastMessagesRead: Long,
                            startMs: Long,
                            endMs: Long,
-                           dateFormat: SimpleDateFormat) = {
+                           dateFormat: SimpleDateFormat,
+                           beginTime: Option[Long] = None,
+                           joinTimeOpt: Option[Long] = None,
+                           periodicJoinTimeInMs: Option[Long] = None) = {
     val elapsedMs: Double = endMs - startMs
     val totalMBRead = (bytesRead * 1.0) / (1024 * 1024)
     val mbRead = ((bytesRead - lastBytesRead) * 1.0) / (1024 * 1024)
-    println("%s, %d, %.4f, %.4f, %d, %.4f".format(dateFormat.format(endMs), id, totalMBRead,
-        1000.0 * (mbRead / elapsedMs), messagesRead, ((messagesRead - lastMessagesRead) / elapsedMs) * 1000.0))
+    print("%s, %d, %.4f, %.4f, %d, %.4f".format(dateFormat.format(endMs), id, totalMBRead,
+      1000.0 * (mbRead / elapsedMs), messagesRead, ((messagesRead - lastMessagesRead) / elapsedMs) * 1000.0))
+    joinTimeOpt match {
+      case Some(joinTime) =>
+        val fetchTimeMs = endMs - beginTime.getOrElse(startMs) - joinTime
+        println(", %d, %d, %d, %.4f, %.4f".format(
+          periodicJoinTimeInMs.getOrElse(0L), joinTime, fetchTimeMs, (bytesRead * 1.0 / (1024 * 1024) / (fetchTimeMs / 1000.0)),
```

Comment:
Why do we have two join times here? We only have one join time in the header, right? The periodic time should be enough I think.